pkg/controller/ingress/concurrency_watchdog.go

```diff
@@ -205,7 +208,9 @@ func (c *ConcurrencyWatchdog) tick(ctx context.Context) error {
 			// don't add the error to return since we shouldn't retry right away
 		}
 	}
-
+	if retErr.ErrorOrNil() != nil {
+		c.logger.Error(retErr, "error reconciling ingress controller resources")
```

Comment:
this would probably be better as

```Go
if err := retErr.ErrorOrNil(); err != nil {
	c.logger.Error(err, "error reconciling ingress controller resources")
        return err
}

return nil
```

Avoids the double ErrorOrNil call and follows Go conventions.

Reply:
changed!